src/System.Linq/src/System/Linq/Iterator.cs

```diff
@@ -20,6 +20,8 @@ protected Iterator()
                 _threadId = Environment.CurrentManagedThreadId;
             }
 
+            protected bool RunningOnCreatingThread => _threadId == Environment.CurrentManagedThreadId;
```

Comment:
I don't think it is after the last round of changes, so it can be reverted. I'm not having a lot of time for CoreFX right now, but I'll try to tidy this up ASAP.